plugins/modules/net_tools/nsupdate.py

```diff
@@ -427,7 +427,10 @@ def ttl_changed(self):
         if lookup.rcode() != dns.rcode.NOERROR:
             self.module.fail_json(msg='Failed to lookup TTL of existing matching record.')
 
-        current_ttl = lookup.answer[0].ttl
+        if self.module.params['type'] == 'NS:
+            current_ttl = lookup.answer[0].ttl if lookup.answer else lookup.authority[0].ttl
```

Comment:
Not sure what exactly the syntax error is, but maybe this helps:
```suggestion
            current_ttl = (lookup.answer[0].ttl if lookup.answer else lookup.authority[0].ttl)
```